superset/datasets/dao.py

```diff
@@ -85,12 +85,22 @@ def validate_table_exists(
             return False
 
     @staticmethod
-    def validate_uniqueness(database_id: int, schema: Optional[str], name: str) -> bool:
+    def validate_uniqueness(
+        database_id: int,
+        schema: Optional[str],
+        name: str,
+        dataset_id: Optional[int] = None,
+    ) -> bool:
         dataset_query = db.session.query(SqlaTable).filter(
             SqlaTable.table_name == name,
             SqlaTable.schema == schema,
             SqlaTable.database_id == database_id,
         )
+
+        if dataset_id:
+            # Validate that the dataset is not referencing itself
```

Comment:
Nit, we actually validate if the dataset is referencing itself — not that we return a `not` in line 104, ie, the inverse of the query.

```suggestion
            # make sure the dataset found is different from the target (if any)
```